.github/workflows/merge-main.yml

```diff
@@ -286,8 +286,16 @@ jobs:
             -p PROMOTE=${{ github.repository }}/${{ matrix.name }}:test
             ${{ matrix.parameters }}
 
-      - name: Promote Image to PROD
-        uses: shrink/actions-docker-registry-tag@v3
+  image-promotions:
+    name: Promote Images to PROD
+    needs:
+      - deploy-prod
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        component: [backend, database]
+    steps:
+      - uses: shrink/actions-docker-registry-tag@v3
         with:
           registry: ghcr.io
           repository: ${{ github.repository }}/${{ matrix.component }}
```

Comment:
Could the problem be that we're using *component*, which isn't defined in the matrix, instead of *name*?